js/queen.js

```diff
@@ -107,7 +136,11 @@ export class Queen extends Piece{
          * Get row decreasing and col increasing moves
          */
         for (let i = this.rowPos - 1, y = this.colPos + 1; i >= minRows && y < maxCols; i--, y++) {
-            if (board[i][y] !== undefined && board[i][y].getPlayerId() == this.playerId) {
+            if (board[i][y] && board[i][y].getPlayerId() == this.playerId) {
```

Comment:
`===` should be used instead of `==` on line `139`.

Reply:
Updated.